### DISCUSSION TITLE
Cloudcompare .obj suitability?

### DISCUSSION THREAD

**Discussion by Pilbarian** [Stellarium/stellarium#3660] [Q&A]
Hi. I have a Lidar data set in Cloudcompare that I can generate a Wavefront .obj file from. Would that be suitable for attempting to create a 3d scene from? Very new to this but I would like to have a crack at creating a 3d scene. The data has georeferencing UTM. Cheers

**Comment by gzotti** [comment]
In principle, yes. I have never used CC for meshing, does that work well? Scenery3D needs a mesh, and if you want detailed colors, you need to transfer any vertex colors from OBJ's "v x y z r g b" lines into the texture. The mesh should not be much larger than about 10 million triangles. I have used 14 OK, but it starts to become slow then. Depends on hardware of course. Reduce regionally, in less interesting areas, obviously.
Then there is a problem with UTM coordinates. CC uses the same offset solution as Stellarium does. If you can export the OBJ in the reduced coordinate space, just configure the large-coordinate offset (local origin) in the scenery3d.ini.
  
If you found out a usable workflow, we are always happy to add such information for the User Guide. If you want then to share your scene and it is of wider interest, we can host such scenes.

**Reply by Pilbarian** [reply]
Thank you. 
I will begin experimenting on my days off work next week. CC does allow for coordinate scaling at the import stage of working with the Lidar data set, I imagine that is what you may be referring to by “offset”? It is easy to cut unwanted (unseen) areas from the mesh to reduce its size so I will do that.
Thanks for your reply.

**Reply by gzotti** [reply]
Not coordinate scaling (shrinking scale), but shifting to a domain where you can express millimetres in single-precision floats. When the mesh uses full UTM coordinates, you will likely lose spatial resolution.  If you see vertex coordinates like 593829.12345/5432101.54321, reduce them to 829.12345/101.54321 and use orig_E=593000, orig_N=5432000 and possibly orig_H, see User Guide, ch.15.   Basically, if Meshlab displays it well, so should Stellarium.

**Reply by Pilbarian** [reply]
Thank you. Great help. Will let you know how I go.

**Reply by gzotti** [reply]
@Pilbarian any success stories to share?
